### PR TITLE
fix: fastify v3 compatibility

### PR DESCRIPTION
**Pull request by SimenB**
For #35. Not including the test change that issue mentions as v3 is not stable yet, but I think this should work for both v2 and v3. I can confirm it works for v3 RC 4 (latest as of time of writing)

### CODE REVIEW

**Comment by puzpuzpuz** [comment]
Thanks for submitting this PR! I'll leave it in "waiting for review" state until Fastify v3 is released. Once this happens, it would make sense to have tests for both v2 and v3.

**Comment by puzpuzpuz** [comment]
@SimenB as Fastify v3 is now released, we can proceed with this PR. Could you update tests, so that they run against Fastify v2 and v3? It should be similar to how it's done for koa v1, but we can reuse the same test script as in case of Fastify the API is the same. It would be also nice to update documentation, so it mentions both v2 and v3.

**Comment by SimenB** [comment]
@puzpuzpuz updated. I removed the `.use` test for v3 since it requires some extra plugins for compat with the `connect` pattern

File: tests/fastify.test.js [hunk 1]
```diff
@@ -6,329 +6,310 @@ const request = require('supertest')
...
-      throw new Error('Unexpected type')
-  }
-}
-
-for (const type of types) {
-  describe(`cls-rtracer for Fastify - ${type}`, () => {
-    test('does not return id outside of request', () => {
-      const id = rTracer.id()
-      expect(id).toBeUndefined()
+describe('cls-rtracer for Fastify', () => {
```
Comment (SimenB) [inline tests/fastify.test.js:9]: bad diff... just ignore whitespace

File: tests/fastifyv2.test.js [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+      return app.register(rTracer.fastifyPlugin, options)
+    case middlewareType:
+      return app.use(rTracer.fastifyMiddleware(options))
+    default:
+      throw new Error('Unexpected type')
+  }
+}
+
+for (const type of types) {
+  describe(`cls-rtracer for Fastify@2 - ${type}`, () => {
```
Comment (puzpuzpuz) [inline tests/fastifyv2.test.js:25]: nit: make test name more consistent with koa v1 one
```suggestion
  describe(`cls-rtracer for Fastify v2 - ${type}`, () => {
```

File: README.md [hunk 3]
```diff
@@ -9,7 +9,7 @@ Request Tracer - Express & Koa middlewares and Fastify & Hapi plugins for CLS-ba
 
 Automatically generates a UUID V1 value as the id for each request and stores it in `AsyncLocalStorage` (CLS core API, see [this blog post](https://itnext.io/one-node-js-cls-api-to-rule-them-all-1670ac66a9e8)). Optionally, if the request contains `X-Request-Id` header, uses its value instead. Allows to obtain the generated request id anywhere in your routes later and use it for logging or any other purposes.
 
-Tested and works fine with Express v4, Fastify v2, Koa (both v1 and v2), and Hapi v18.
+Tested and works fine with Express v4, Fastify v2 and v3, Koa v1 and v2, and Hapi v18.
```
Comment (puzpuzpuz) [inline README.md:12]: Can't comment directly, so going to leave the suggestion here.

L132 (Legacy Fastify middleware section): Change `There is a connect-style middleware available for Fastify` to `There is a connect-style middleware available for Fastify v2`.

**Review by puzpuzpuz — APPROVED** [review]
Well done. Left a couple of documentation related comments.

**Comment by puzpuzpuz** [comment]
@SimenB many thanks for your contribution!

**Pull request merged**